tests/output.rs

```diff
@@ -714,3 +717,333 @@ fn test_overlapping_matches_vimgrep() {
         "#,
     );
 }
+
+#[test]
+fn test_after_context() {
+    assert_sorted_output(
+        "rust_project",
+        r#"
+            $ tree-sitter-grep -q '(function_item) @f' -l rust --after-context 2
+            src/stop.rs:1:fn stop_it() {}
+            src/helpers.rs:1:pub fn helper() {}
+            src/lib.rs:3:pub fn add(left: usize, right: usize) -> usize {
+            src/lib.rs:4:    left + right
+            src/lib.rs:5:}
+            src/lib.rs-6-
+            src/lib.rs-7-#[cfg(test)]
+            --
+            src/lib.rs:12:    fn it_works() {
+            src/lib.rs:13:        let result = add(2, 2);
+            src/lib.rs:14:        assert_eq!(result, 4);
+            src/lib.rs:15:    }
+            src/lib.rs-16-}
+            src/lib.rs-17-
+        "#,
+    );
+}
+
+#[test]
+fn test_after_context_matches_overlap_context_lines() {
+    assert_sorted_output(
+        "rust_overlapping",
+        r#"
+            $ tree-sitter-grep -q '(call_expression function: (identifier) @function_name (#match? @function_name "^h"))' -l rust -A 2
+            src/lib.rs:10:    hello();
+            src/lib.rs:11:    hoo();
+            src/lib.rs-12-    raa();
+            src/lib.rs-13-    roo();
+        "#,
+    );
+}
+
+#[test]
+fn test_after_context_overlapping_matches() {
+    assert_sorted_output(
+        "rust_overlapping",
+        r#"
+            $ tree-sitter-grep -q '(closure_expression) @c' -l rust --after-context 2
+            src/lib.rs:2:    let f = || {
+            src/lib.rs:3:        || {
+            src/lib.rs:4:            println!("whee");
+            src/lib.rs:5:        }
+            src/lib.rs:6:    };
+            src/lib.rs-7-}
+            src/lib.rs-8-
+        "#,
+    );
+}
+
+#[test]
+fn test_after_context_overlapping_multiline_matches_vimgrep() {
+    assert_sorted_output(
+        "rust_overlapping",
+        r#"
+            $ tree-sitter-grep -q '(closure_expression) @c' -l rust --after-context 2 --vimgrep
+            src/lib.rs:2:13:    let f = || {
+            src/lib.rs:3:9:        || {
+            src/lib.rs-7-}
```

Comment:
This is kind of weird that there's a "gap" between the match lines printed by `--vimgrep` (which only outputs the first line of each match) and then the context lines (which are the lines after the actual end of the matched AST nodes)

But it does in fact appear to match the corresponding `ripgrep` behavior (if you combine a `--multiline` search whose matches do in fact span multiple lines with `--context`/`--after-context` and `--vimgrep` you also see such a "gap")